+nosnoc/+discrete_time_problem/Heaviside.m

```diff
@@ -399,36 +456,113 @@ function generate_direct_transcription_constraints(obj)
                     model.x_ref_end_val,...
                     p_global);
             end
+
+            %  -- Constraint for the terminal numerical and physical time (if no equidistant grids are required) --
```

Comment:
This part is quite complicated as there are many possible cases.
maybe it would be useful to write a single line comment after every if/else to summarize what is happening :D
otherwise one needs to think quite a bit